### PR TITLE
Solaris 10

### PR DESCRIPTION
**Pull request by pekdon**
A few compile fixes for Solaris 10.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -173,6 +173,15 @@ if test "$os_unix" = "yes"; then
 	                AC_MSG_RESULT([yes])],
 	               [AC_MSG_RESULT([no])])
 
+    # Check if SUN_LEN is defined
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <sys/un.h>]],
```
[the remark this answers is not in the document]
Comment (pekdon) [reply configure.ac:177]: It should be sufficient if sys/un.h is included in compat or always included before.

[the remark this answers is not in the document]
Comment (pekdon) [reply configure.ac]: Updated the PR branch with that change

**Comment by pekdon** [comment]
Ok with this PR or are you waiting for me to provide something else?

File: configure.ac (hunk 1 above)
[the remark this answers is not in the document]
Comment (pekdon) [reply configure.ac]: @ueno any more feedback on this one, something you want me to fix?

[the remark this answers is not in the document]
Comment (ueno) [reply configure.ac]: Thank you for the update and apologies for the delay; I somehow missed the notification.

**Review by ueno — APPROVED** [review]

File: common/compat.h [hunk 2]
```diff
@@ -39,6 +39,7 @@
 
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/un.h>
```
Comment (ueno) [inline common/compat.h:42]: In the mingw CI:
```console
../common/compat.h:42:10: fatal error: sys/un.h: No such file or directory
   42 | #include <sys/un.h>
      |          ^~~~~~~~~~
```
I suppose a configure check is nevertheless needed :-(

Comment (pekdon) [reply common/compat.h:42]: Ok, I'll update the PR

**Comment by ueno** [comment]
Thank you!

**Pull request merged**